src/kit.ts

```diff
@@ -779,25 +847,28 @@ const VsGenerators: {[key: string]: string} = {
 async function varsForVSInstallation(inst: VSInstallation, hostArch: string, targetArch?: string): Promise<Map<string, string>|null> {
   console.log(`varsForVSInstallation path:'${inst.installationPath}' version:${inst.installationVersion} host arch:${hostArch} - target arch:${targetArch}`);
   const common_dir = path.join(inst.installationPath, 'Common7', 'Tools');
+  const majorVersion = parseInt(inst.installationVersion);
+
   let vcvarsScript: string = 'vcvarsall.bat';
-  if (targetArch === "arm" || targetArch === "arm64") {
-    // The arm(64) vcvars filename for x64 hosted toolset is using the 'amd64' alias.
-    vcvarsScript = `vcvars${kitHostTargetArch(hostArch, targetArch, true)}.bat`;
+  if (majorVersion < 14) {
+    if (targetArch === "arm" || targetArch === "arm64") {
```

Comment:
@lygstate, did you have anything in mind with moving the whole targetArch === "arm" or "arm64" inside the if majorVersion < 14? It causes arm/arm64 kits to be created in cmake-user-kits.json even when arm/arm64 is not installed.

Reply:
Indeed, I don't know why this is needed, as for  majyorVersion>=14, we can always using 'vcvarsall.bat'.
And for vs2010, 'vcvarsall.bat' also exist. I even think vcvarsScript = `vcvars${kitHostTargetArch(hostArch, targetArch, true)}.bat`; should be removed totally, I don't know why this present

Reply:
The reason for invoking vcvars target custom name is that, when arm/arm64 is not installed, that script will not exist. If installed, the script will exist. If we use vcvarsall, it exists regardless of arm/arm64 installed and also it doesn't fail when run. On top of that, it produces INCLUDE and other env vars so it will cause arm/arm64 kits to be created, listed even if there is no arm/arm64 support in the VS installation. Users can be very confused by this.